flume-ng-core/src/main/java/org/apache/flume/source/AvroSource.java

```diff
@@ -292,12 +287,7 @@ private ChannelPipelineFactory initChannelPipelineFactory() {
         keystorePassword, keystoreType, enableIpFilter,
         patternRuleConfigDefinition);
     } else {
-      pipelineFactory = new ChannelPipelineFactory() {
-        @Override
-        public ChannelPipeline getPipeline() throws Exception {
-          return Channels.pipeline();
-        }
-      };
+      pipelineFactory = () -> Channels.pipeline();
```

Comment:
This can be replaced by method reference, i.e. `pipelineFactory = Channels::pipeline`